src/material.js

```diff
@@ -173,20 +173,42 @@ export class Material {
     }
 
     /**
-     * Calculates hash from basis and lattice. Algorithm expects the following:
+     * Calculates the hash for a material.
+     * If `isNonPeriodic` = false
+     * The hash is calculated from the basis and lattice. The algorithm expects the following:
      * - asserts lattice units to be angstrom
      * - asserts basis units to be crystal
      * - asserts basis coordinates and lattice measurements are rounded to hash precision
      * - forms strings for lattice and basis
      * - creates MD5 hash from basisStr + latticeStr + salt
+     *
+     * If `isNonPeriodic` = true
+     * The hash based on the `inchi` string for the material that is returned from getNonPeriodicHashMessage.
+     *
      * @param salt {String} Salt for hashing, empty string by default.
      * @param isScaled {Boolean} Whether to scale the lattice parameter 'a' to 1.
      */
     calculateHash(salt = '', isScaled = false) {
-        const message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        let message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        if (this._json.isNonPeriodic) {
+            message = this.getNonPeriodicHashMessage() || message;
+        }
         return CryptoJS.MD5(message).toString();
     }
 
+    /**
+     * Gets the `inchi` string from the derived properties and returns it as the message to use for
+     * generating the hash when `isNonPeriodic` = true.
+     * @returns String
+     */
+    getNonPeriodicHashMessage() {
+        const derivedProperties = this._json.derivedProperties;
+        const inchi = lodash.isArray(derivedProperties) ? derivedProperties.find(x => x.name === "inchi") : null;
```

Comment:
We are facing a conceptual problem here - `derivedProperties` and their handling seem to not be defined in made.js by design previously. We need to consider why and how the implementation needs to be adjusted.